controller/Lists.php

```diff
@@ -64,38 +63,44 @@
 class Lists extends tao_actions_CommonModule
 {
     use OntologyAwareTrait;
+    use HttpJsonResponseTrait;
 
     private const REMOTE_LIST_PREVIEW_LIMIT = 20;
 
     /** @var bool */
     private $isListsDependencyEnabled;
 
     /**
-     * Show the list of users
+     * This REST endpoint:
+     * - Returns the page with the lists for GET requests
+     * - Creates a new list and returns its name and URI for POST requests
+     *
      * @return void
      */
     public function index()
     {
-        tao_helpers_Scriptloader::addCssFile(Template::css('lists.css', 'tao'));
-        $this->defaultData();
-
-        $myAdderFormContainer = new tao_actions_form_List();
-        $myForm = $myAdderFormContainer->getForm();
-
-        if ($myForm->isSubmited()) {
-            if ($myForm->isValid()) {
-                $values = $myForm->getValues();
-                $newList = $this->getListService()->createList($values['label']);
-                $i = 0;
-                while ($i < $values['size']) {
-                    $this->getListService()->createListElement($newList, __('element') . ' ' . ($i + 1));
-                    $i++;
-                }
+        if ($this->getPsrRequest()->getMethod() == 'POST') {
+            if (!$this->isXmlHttpRequest()) {
+                throw new common_exception_BadRequest('wrong request mode');
             }
-        } else {
-            $myForm->getElement('label')->setValue(__('List') . ' ' . (count($this->getListService()->getLists()) + 1));
+
+            $newName = __('List') . ' ' . (count($this->getListData()) + 1);
+            $list = $this->getListService()->createList($newName);
+            $this->getListService()->createListElement($list, __('element') . ' 1');
+
+            $createdResponse = [
+                'newList' => [
+                    'name' => $newName,
+                    'uri' => $list->getUri(),
+                ]
+            ];
```

Comment:
Goo job! :) 

It is totally optional, but I would like to see your point of view on this: 

1) The logic still remains in the controller. Do you think it makes sense to move this logic to an application service and keep the controller acting as just an intermediary?

Example: 

```
class ListCreator
{
    /**
     * @throws common_exception_BadRequest
     */
    public function createByRequest(ServerRequest $request): List;    
}
```

And the List (example) object could have a `jsonSerialize(): array` method to be used here.

2) Do you think it is needed to name the attribute as `newList` since it is returned in a POST + 201 endpoint?

Reply:
1. I like the idea of having a service to encapsulate the creation for the list itself.

However, since we don't have an explicit "application" namespace, right now I'm not sure where to place it. @gabrielfs7 @shpran Do you think it be ok to have it on a nested namespace inside the controller namespace (`oat\taoBackOffice\controller\service`) or do you think it would be better to keep it easy put it into the `model` namespace (as, for example, MediaManager does for `editInstanceForm`, `FileImporter`, etc)?

2. Yes, the created resource is returned nested on another attribute of the root object returned by the response (instead of returning the created resource directly). At first I put it this way in case we need to also return data for the item created inside that list (i.e. its ID and name), so it would have something like `{newList: {....}, newItem: {....}}` but, as you suggested, is better to return it directly

Reply:
For the point 1, indeed we do not have "yet" layer separation, but I would not go into create anything but controllers in the controllers namespace. 

I think you can find some namespace inside  the `models` namespace instead. I.e: `oat\taoBackOffice\model\lists\Service`

Reply:
Changed :point_right:  8bd0c456b7267cfa190c2b3384e080e584baab0e

Reply:
:exploding_head: missed that one. Then I'll be adding it to a new `ListCreator` inside `model\lists`

Reply:
I've pushed some draft changes for that one in d55ae4532fd530ae9e4fc5b87bf14c833ae5218e, please take a look. I'll come back to this pull request when I'm done with the backport.